File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.StringTokenizer;
import java.util.stream.Collectors;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringBuffer sb = new StringBuffer();

        int n = Integer.parseInt(br.readLine());
        Map<String, String> map = new HashMap<>();

        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String name =  st.nextToken();
            String history =  st.nextToken();

            if (history.equals("enter")) {
                map.put(name, history);
            } else {
                map.remove(name);
            }
        }

        List<String> nameList = new ArrayList<>(map.keySet());
        nameList = nameList.stream().sorted().collect(Collectors.toList());
        for(int i = nameList.size()-1; i >= 0; i--) {
            sb.append(nameList.get(i)).append("\n");
        }
        bw.write(sb.toString().trim());
        bw.flush();
        bw.close();
    }
}

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val sb = StringBuilder()

    val n = br.readLine().toInt()
    var map = HashMap<String, String>()
    for (i in 0 until n) {
        val input = br.readLine()
        val name = input.split(" ")[0]
        val history = input.split(" ")[1]

        if (history.equals("leave")) {
            map.remove(name)
        } else {
            map.put(name, history)
        }
    }

    val nameList = map.keys.sortedDescending().toList()
    for (name in nameList) {
        sb.append(name).append("\n")
    }

    bw.write(sb.toString())
    bw.flush()
    bw.close()
}